tests/framework/results/test_context.py

```diff
@@ -443,3 +445,130 @@ def test__bad_aggregator_stratification():
     with pytest.raises(KeyError, match="height"):
         for result, _measure in ctx.gather_results(population, event_name):
             print(result)
+
+
+@pytest.mark.parametrize("pop_filter", ['familiar=="spaghetti_yeti"', ""])
+def test__filter_population(pop_filter):
+    filtered_pop = ResultsContext()._filter_population(
+        population=BASE_POPULATION, pop_filter=pop_filter
+    )
+    if pop_filter:
+        # The pop filter is set up to filter out everyone
+        assert filtered_pop.empty
+    else:
+        # An empty pop filter should return the entire population
+        assert filtered_pop.equals(BASE_POPULATION)
+
+
+@pytest.mark.parametrize("stratifications", [("familiar",), ()])
+def test__get_groups(stratifications):
+    groups = ResultsContext()._get_groups(
+        stratifications=stratifications, filtered_pop=BASE_POPULATION
+    )
+    assert isinstance(groups, DataFrameGroupBy)
+    if stratifications:
+        # Check that all familiars exist
+        assert set(groups.groups.keys()) == set(FAMILIARS)
+        # Check that the entire population is included
+        assert sum([len(value) for value in groups.groups.values()]) == len(BASE_POPULATION)
+    else:
+        item = groups.groups.popitem()
+        # Check that there are no other groups
+        assert not groups.groups
+        # Check that the group is 'all' and includes the entire population
+        key, val = item
+        assert key == "all"
+        assert val.equals(BASE_POPULATION.index)
+
+
+@pytest.mark.parametrize(
+    "stratifications, aggregator_sources",
+    [
+        (("familiar",), ["power_level"]),
+        (("familiar",), []),
+        (("familiar", "house"), ["power_level"]),
+        (("familiar", "house"), []),
+        ((), ["power_level"]),
+        ((), []),
+    ],
+)
+def test__aggregate(stratifications, aggregator_sources):
+    """Test that we are aggregating correctly. There are some nuances here:
+      - If aggregator_resources is provided, then simply .apply it to the groups passed in.
+      - If no aggregator_resources are provided, then we want a full aggregatio of the groups.
+
+    Note that the groups can be either a pandas DataFrame or a DataFrameGroupBy.
+    """
+    groups = ResultsContext()._get_groups(
+        stratifications=stratifications, filtered_pop=BASE_POPULATION
+    )
+    aggregates = ResultsContext()._aggregate(
+        pop_groups=groups,
+        aggregator_sources=aggregator_sources,
+        aggregator=len,
+    )
+    if stratifications:
+        full_idx = (
```

Comment:
Nit: `full_idx` suggests the full pop table index to me. Maybe `stratification_idx`?